src/main/java/doit/jpastudy2/repository/Category.java

```diff
@@ -15,19 +15,24 @@
 public class Category {
 
     @Id // PK임을 나타낸다.
-    @GeneratedValue(strategy = GenerationType.AUTO) // 자동 생성되는 값임을 나타낸다.
-    @Column(name = "category_id") // 컬럼명을 지정한다.
-    private Long id;
+    //@GeneratedValue(strategy = GenerationType.AUTO) // 자동 생성되는 값임을 나타낸다.
+    @Column(name = "exercise") // 컬럼명을 지정한다.
+    private String exercise;
```

Comment:
우선 수고하셨습니다! 코드 보니 어떠한 어려움이 있으셨을지 예상이 가서 몇가지 정보를 공유해드리겠습니다.

1. Entity에서 @Id는 해당 필드가 PK값임을 의미합니다. 따라서 유일하게 작성될 수 있도록 수동으로 작성하지 않는 편이 좋습니다.
2. 숫자형 타입의 필드에 @Id와 @GeneratedValue(strategy=GenerationType.AUTO)를 함께 사용하게 되면, 데이터베이스 단에서 자동으로 유일한 PK 값을 할당합니다. ( MySQL의 경우 이전의 PK 값 + 1)
3. String 타입의 @Id 필드에는 @GeneratedValue(strategy=GenerationType.AUTO)를 사용할 수 없습니다. 데이터베이스 단에서 이 값을 자동으로 생성할 수 없기 때문입니다.

따라서 아래처럼 작성하는 편이 조금 더 보편적입니다.
```suggestion
    @Id // PK임을 나타낸다.
    @GeneratedValue(strategy = GenerationType.AUTO) // 자동 생성되는 값임을 나타낸다.
    @Column(name = "exercise_id") // 컬럼명을 지정한다.
    private Long id;
```